source/Missions/Services/Agents/Packets/AgentData.cs

```diff
@@ -35,6 +35,40 @@ public AgentData(Agent agent)
             }
         }
 
+        public AgentData(
+            Vec3 position,
+            Vec2 movementDirection,
+            Vec3 lookDirection,
+            Vec2 inputVector,
+            AgentEquipmentData agentEquipment,
+            AgentActionData agentActionData,
+            AgentMountData agentMountData)
+        {
+            Position = position;
+            MovementDirection = movementDirection;
+            LookDirection = lookDirection;
+            InputVector = inputVector;
+            AgentEquipment = agentEquipment;
+            ActionData = agentActionData;
+            MountData = agentMountData;
+        }
+
+        public AgentData(Agent agent, Vec2 movementDirection) : 
```

Comment:
For this we want any changed data to accumulate before sending the agent data. This currently only sends 1 piece at a time.

For example
If lookDirection and movementDirection change before the tick rate we want those in the same packet.